app/controllers/guides_controller.rb

```diff
@@ -25,17 +25,14 @@ def new
   def create
     @guide = Guide.new(guide_params)
 
-    ActiveRecord::Base.transaction do
-      if @guide.save
-        GuidePublisher.new(guide: @guide).put_draft
-        redirect_to edit_guide_path(@guide), notice: "Guide has been created"
-      else
-        render action: :new
-      end
+    publication = Publisher.new(content_model: @guide).
+                            save_draft(GuidePresenter.new(@guide, @guide.latest_edition))
+    if publication.success?
+      redirect_to edit_guide_path(@guide), notice: 'Guide has been created'
+    else
+      flash.now[:error] = publication.errors
```

Comment:
`flash.now` is only available to the current action.. Useful for when you're not redirecting a user and rerendering a template after a failed save/update/summat.

Using `flash[:error]` without the `now` bit is the cause of those random flash messages appearing when you wouldn't expect them if you've seem them lurking :ghost: